plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/SpringChangedHub.java

```diff
@@ -0,0 +1,267 @@
+import org.hotswap.agent.plugin.spring.scanner.ClassChangeEvent;
+import org.hotswap.agent.plugin.spring.utils.ResourceUtils;
+import org.hotswap.agent.util.spring.util.ObjectUtils;
+import org.springframework.beans.PropertyValue;
+import org.springframework.beans.factory.config.*;
+import org.springframework.beans.factory.support.AbstractBeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanNameGenerator;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.context.annotation.AnnotationBeanNameGenerator;
+
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SpringChangedHub implements SpringListener<SpringEvent> {
+    private static AgentLogger LOGGER = AgentLogger.getLogger(SpringChangedHub.class);
+    private static final int DEFAULT_DELAY_PERIOD = 1500;
+    public static final int INIT = 1;
+    public static final int CHANGING = 2;
+    public static final int RELOADED = 3;
+
+    static int maxWaitTimes = 5;
+
+    private static final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1, new ThreadFactory() {
+
+        @Override
+        public Thread newThread(Runnable r) {
+            return new Thread(r, "SpringReload");
+        }
+    });
+
+
+    private AtomicInteger status = new AtomicInteger(1);
+    private AtomicInteger waitTimes = new AtomicInteger(0);
+
+    private DefaultListableBeanFactory defaultListableBeanFactory;
+    private static ClassLoader appClassLoader;
+    private static Map<DefaultListableBeanFactory, SpringChangedHub> springChangedHubs = new ConcurrentHashMap<>(2);
+    final BeanFactoryAssistant beanFactoryAssistant;
+    final AtomicBoolean pause = new AtomicBoolean(false);
+    private SpringReload current;
+    private SpringReload next;
+    private SpringReload failed;
+    private BeanNameGenerator beanNameGenerator = new AnnotationBeanNameGenerator();
+
+
+    public SpringChangedHub(DefaultListableBeanFactory defaultListableBeanFactory) {
+        beanFactoryAssistant = new BeanFactoryAssistant(defaultListableBeanFactory);
+        current = new SpringReload(defaultListableBeanFactory, beanFactoryAssistant);
+        current.setBeanNameGenerator(beanNameGenerator);
+        next = new SpringReload(defaultListableBeanFactory, beanFactoryAssistant);
+        next.setBeanNameGenerator(beanNameGenerator);
+        failed = new SpringReload(defaultListableBeanFactory, beanFactoryAssistant);
+        this.defaultListableBeanFactory = defaultListableBeanFactory;
+        SpringEventSource.INSTANCE.addListener(this);
+    }
+
+    void init() {
+        scheduler.scheduleWithFixedDelay(this::reload, 1, DEFAULT_DELAY_PERIOD, TimeUnit.MILLISECONDS);
```

Comment:
the 'DEFAULT_DELAY_PERIOD' is too long, org.hotswap.agent.command.impl.SchedulerImpl use 100ms as its interval. Or, should we reuse the same mechanism and make 'SpringChangedHub' an instance of MergeableCommand?